pkg/test_framework/helpers.go

```diff
@@ -0,0 +1,240 @@
+	errjson, _ := obj.MarshalJSON()
+	section := obj.UnstructuredContent()
+	for i, v := range path {
+		errpath += "[" + v + "]"
+
+		if i+1 == len(path) {
+			result, ok := section[v]
+			if !ok {
+				return nil, fmt.Errorf("could not find value %s in:\n%s", errpath, string(errjson))
+			} else {
+				return result, nil
+			}
+		} else {
+			tmp, ok := section[v]
+			if !ok {
+				return nil, fmt.Errorf("could not find section %s in:\n%s", errpath, string(errjson))
+			}
+			section, ok = tmp.(map[string]interface{})
+			if !ok {
+				return nil, fmt.Errorf("found %s value, but section was expected in:\n%s", errpath, string(errjson))
+			}
+		}
+	}
+	panic("Unreachable code in ReadValue")
+}
+
+func WriteValue(obj *unstructured.Unstructured, value interface{}, path ...string) error {
+	if len(path) == 0 {
+		return fmt.Errorf("empty path")
+	}
+	errpath := ""
+	errjson, _ := obj.MarshalJSON()
+	section := obj.UnstructuredContent()
+	for i, v := range path {
+		errpath += "[" + v + "]"
+
+		if i+1 == len(path) {
+			section[v] = value
+			return nil
+		} else {
+			tmp, ok := section[v]
+			if !ok {
+				return fmt.Errorf("could not find section %s in:\n%s", errpath, string(errjson))
+			}
+			section, ok = tmp.(map[string]interface{})
+			if !ok {
+				return fmt.Errorf("found %s value, but section was expected in:\n%s", errpath, string(errjson))
+			}
+		}
+	}
+	panic("Unreachable code in WriteValue")
+}
+
+func DeleteValue(obj *unstructured.Unstructured, path ...string) error {
+	if len(path) == 0 {
+		return fmt.Errorf("empty path")
+	}
+	errpath := ""
+	errjson, _ := obj.MarshalJSON()
+	section := obj.UnstructuredContent()
+	for i, v := range path {
+		errpath += "[" + v + "]"
+
+		if i+1 == len(path) {
+			delete(section, v)
+			return nil
+		} else {
+			tmp, ok := section[v]
+			if !ok {
+				return fmt.Errorf("could not find section %s in:\n%s", errpath, string(errjson))
+			}
+			section, ok = tmp.(map[string]interface{})
+			if !ok {
+				return fmt.Errorf("found %s value, but section was expected in:\n%s", errpath, string(errjson))
+			}
+		}
+	}
+	panic("Unreachable code in DeleteValue")
+}
+
+func LoadUnstructured(r io.Reader) (*unstructured.Unstructured, error) {
+	reader, _, isJson := yaml.GuessJSONStream(r, bytes.MinRead)
+	data, err := ioutil.ReadAll(reader)
+	if err != nil {
+		return nil, err
+	}
+	if !isJson {
+		tmp, err := yaml.ToJSON(data)
+		if err != nil {
+			return nil, err
+		}
+		data = tmp
+	}
+
+	result := unstructured.Unstructured{}
+	err = result.UnmarshalJSON(data)
+	return &result, err
+}
+
+func LoadInto(r io.Reader, into interface{}) error {
+	if err := yaml.NewYAMLOrJSONDecoder(r, bytes.MinRead).Decode(into); err != nil {
+		return err
+	}
+	return nil
+}
+
+type WaitSource func()(runtime.Object, error)
+
+func WaitForNone(timeout time.Duration, from WaitSource) error {
+	none := int32(0)
+	return WaitFor(&none, timeout, from)
+}
+
+func WaitFor(reps* int32, timeout time.Duration, from WaitSource) error {
```

Comment:
Would it be useful to also have a [`WaitForResource`](https://sourcegraph.yelpcorp.com/services/kafka-operator/-/blob/acceptance/helpers.go#L82:6) helper function?

Reply:
Honestly, I am not sure. Considering how you use that function in kafka operator acceptance tests, it is not clear to me at this moment what useful generalisation could look like. But **maybe** I will figure out something later.